src/js/components/uploadDetachedFiles/uploadDetachedFileValidation.jsx

```diff
@@ -0,0 +1,284 @@
+		else if (data.jobs[0].job_status == 'finished') {
+			// don't run the check again if it's done
+			runCheck = false;
+
+			// display dowload buttons
+			// make a clone of the file's react state
+			const item = Object.assign({}, this.state.detachedAward);
+			item.status = "done";
+
+			ReviewHelper.validateDetachedSubmission(this.props.submission.id)
+				.then((response) => {
+					this.setState({
+						detachedAward: item,
+						validationFinished: true,
+						headerErrors: false,
+						jobResults: response,
+						published: data.publish_status
+					});
+				});
+		}
+
+		if (this.isUnmounted) {
+			return;
+		}
+
+		if (runCheck && !this.isUnmounted) {
+			// wait 5 seconds and check the file status again
+			window.setTimeout(() => {
+				this.checkFileStatus(this.props.submission.id);
+			}, timerDuration * 1000);
+		}
+	}
+
+	submitFabs(){
+		UploadHelper.submitFabs({'submission_id': this.props.submission.id})
+			.then((response)=>{
+				this.setState({submit: false})
+			})
+			.catch((error)=>{
+				if(error.httpStatus == 400){
+					this.setState({error: 1, submit: false});
+				}else if(error.httpStatus == 500){
+					this.setState({error: 3, submit: false});
+				}
+			})
+	}
+
+	// ERRORS
+	// 1: Submission is already published
+	// 2: Fetching file metadata failed
+	// 3: File already has been submitted in another submission
+
+	render() {
+		let validationButton = null;
+		let validationBox = null;
+		let agency = this.state.agency;
+		let startDate = this.state.rep_start;
+		let endDate = this.state.rep_end;
+
+		let headerDate = null;
+		if(this.state.agency != '' && this.state.rep_start != '' && this.state.rep_end != ''){
+			headerDate = <div className="col-md-2 ">
+										<div className = 'header-box'>
+												<span>
+												Agency: {this.state.agency}
+												</span>
+												<br/>
+												<span>
+												Date: {startDate} - {endDate}
+												</span>
+											</div>
+									</div>;
+		}
+
+		const type = {
+			fileTitle: 'Upload',
+			fileTemplateName: 'detached_award.csv',
+			requestName: 'detached_award',
+			progress: '0'
+		}
+		validationBox = <ValidateDataFileContainer type={type} data={this.state.jobResults}/>;
+		if(!this.state.headerErrors && this.state.validationFinished) {
+			validationBox = <ValidateValuesFileContainer type={type} data={this.state.jobResults} />;
+			if(this.state.jobResults.detached_award.error_type == "none" && this.state.error == 0) {
+				validationButton = <button className='pull-right col-xs-3 us-da-button' onClick={this.submitFabs.bind(this)}>Publish</button>;
+				if(this.state.published == 'published'){
+					validationButton = <button className='pull-right col-xs-3 us-da-disabled-button' onClick={this.submitFabs.bind(this)} disabled>File Already Published</button>;
+				}
+			}
+		}
+
+		let errorMessage = null;
+		let errorText = null;
+		let errorDesc = null;
+		if (this.state.detachedAward.error.show || this.state.error != 0) {
```

Comment:
Same as previous file - let's move this logic out of `render` into it's own function